jsapp/js/projects/projectsTable/colResize.tsx

```diff
@@ -0,0 +1,271 @@
+          });
+        }
+        // Set variables for de-duping
+        dragPrevXRef.current = e.pageX; // to skip event early if same x
+        dragPrevWidthRef.current = newWidth; // to skip state update if same width
+        return;
+      }
+
+      // Mousedown
+      //  - Start the dragging interaction (if on a resize handle)
+      if (
+        e.type === 'mousedown' &&
+        e.button === 0 // Only on left (primary) mouse button
+      ) {
+        // Detect resize handle with [data-resize-fieldname={fieldname}]
+        const fieldname = (e.target as HTMLElement).dataset.resizeFieldname;
+        if (fieldname) {
+          setShouldRenderBodyCursor(true);
+          isDraggingRef.current = true;
+          draggingColumnRef.current = fieldname;
+          dragStartXRef.current = e.pageX;
+          dragPrevXRef.current = e.pageX;
+
+          // Capture the current width of the resizer's parent element, a
+          // header cell, same width as all the row cells below it.
+          const parent = (e.target as HTMLElement).parentElement;
+          if (parent) {
+            dragStartWidthRef.current = parent.offsetWidth;
+            // box-sizing: border-box makes this very easy
+          }
+        }
+        return;
+      }
+
+      // Mouseup (or contextmenu)
+      //  - End the dragging interaction
+      //  - Clear the cursor
+      if (e.type === 'mouseup' || e.type === 'contextmenu') {
+        // // Uncomment to help find good width settings
+        // if (isDraggingRef.current) {
+        //   console.log(
+        //     'resized',
+        //     draggingColumnRef.current,
+        //     'to',
+        //     dragPrevWidthRef.current,
+        //     'px'
+        //   );
+        // }
+        isDraggingRef.current = false;
+        setShouldRenderBodyCursor(false);
+        return;
+      }
+    }
+  });
+  const eventTypes = ['mousedown', 'mouseup', 'mousemove', 'contextmenu'];
+
+  // Mount/Unmount effect: Attach/remove handlers on the body
+  useEffect(() => {
+    // Mount! Add handlers on body
+    eventTypes.forEach((eventType) => {
+      document.body.addEventListener(eventType, handlerRef.current);
+    });
+
+    // Unmount! Remove handlers on body
+    return () => {
+      eventTypes.forEach((eventType) => {
+        document.body.removeEventListener(eventType, handlerRef.current);
+      });
+    };
+  }, []);
+
+  // Log confirms minimal re-rendering.
+  // console.log('Rendered resizer!');
+
+  // Render a dynamic style tag to update all custom widths in a single step,
+  // without modifying any attributes managed elsewhere by React.
+  return (
+    <>
+      <style>
+        {Object.keys(columnWidths).map(
+          (column) =>
+            `
+            [data-field="${column}"] {
+              width: ${columnWidths[column]}px !important;
+              max-width: ${columnWidths[column]}px !important;
+            }
+            `
+        )}
+      </style>
+      <style>
```

Comment:
Do we need separate style tags?

Reply:
Probably not. IIRC I went with it on random speculation that React might have more fun diffing if the one thing is in its own style tag, but I have no reason to believe that. XD